pkg/object/statussynccontroller/statussynccontroller.go

```diff
@@ -192,29 +241,36 @@ func (ssc *StatusSyncController) handleStatus(unixTimestamp int64) {
 }
 
 func (ssc *StatusSyncController) syncStatusToCluster(statuses map[string]string) {
-	kvs := make(map[string]*string)
-
 	// Delete statuses which disappeared in current status.
 	if ssc.lastSyncStatuses != nil {
+		kvs := make(map[string]*string)
 		for k := range ssc.lastSyncStatuses {
 			if _, exists := statuses[k]; !exists {
 				k = ssc.superSpec.Super().Cluster().Layout().StatusObjectKey(k)
 				kvs[k] = nil
 			}
 		}
+		err := ssc.superSpec.Super().Cluster().PutAndDeleteUnderLease(kvs)
+		if err != nil {
+			logger.Errorf("sync status failed. If the message size is too large, "+
+				"please increase the value of cluster.MaxCallSendMsgSize in configuration: %v", err)
+		}
 	}
 
 	ssc.lastSyncStatuses = statuses
 
-	for k, v := range statuses {
-		k = ssc.superSpec.Super().Cluster().Layout().StatusObjectKey(k)
-		kvs[k] = &v
-	}
-
-	err := ssc.superSpec.Super().Cluster().PutAndDeleteUnderLease(kvs)
-	if err != nil {
-		logger.Errorf("sync status failed. If the message size is too large, "+
-			"please increase the value of cluster.MaxCallSendMsgSize in configuration: %v", err)
+	kvs := make(map[string]*string)
+	for k, value := range statuses {
```

Comment:
if `len(kvs) > 0`, should call `PutAndDeleteUnderLease` after the for loop too.

Reply:
Fixed!